src/components/structures/MessagePanel.js

```diff
@@ -296,6 +296,12 @@ module.exports = React.createClass({
             // Wrap consecutive member events in a ListSummary
             if (isMembershipChange(mxEv)) {
                 let ts1 = mxEv.getTs();
+                // Ensure that the key of the MemberEventListSummary does not change with new
+                // member events. This will prevent it from being re-created necessarily, and
+                // instead will allow new props to be provided. In turn, the shouldComponentUpdate
+                // method on MELS can be used to prevent unnecessary renderings.
+                // `prevEvent` at this point is a non-member event or null.
+                const key = "memberlistsummary-" + (prevEvent ? prevEvent.getId() : "");
```

Comment:
> But still, using the prevEvent's eventid as the key feels odd here. How about (prevEvent ? mxEv.getId() : 'initial');. And it needs comments.

This still does not keep the key of `MemberEventListSummary` constant WRT events changing within it. `mxEv` is an event within the summary, which may therefore change. The (non-member) event prior to the summary will not change, whether it's `null` (during back-pagination through a number of membership events) or otherwise.